common/features/transaction/types.ts

```diff
@@ -0,0 +1,106 @@
+import { BroadcastAction } from './broadcast/types';
+import {
+  FieldAction,
+  InputFieldAction,
+  SetToFieldAction,
+  SetValueFieldAction,
+  SetDataFieldAction
+} from './fields/types';
+import { MetaAction, SetTokenToMetaAction, SetTokenValueMetaAction } from './meta/types';
+import { NetworkAction } from './network/types';
+import { SignAction } from './sign/types';
+
+export enum TRANSACTION {
+  CURRENT_VALUE_SET = 'CURRENT_VALUE_SET',
+  CURRENT_TO_SET = 'CURRENT_TO_SET',
+  TOKEN_TO_ETHER_SWAP = 'TOKEN_TO_ETHER_SWAP',
+  ETHER_TO_TOKEN_SWAP = 'ETHER_TO_TOKEN_SWAP',
+  TOKEN_TO_TOKEN_SWAP = 'TOKEN_TO_TOKEN_SWAP',
+  SEND_EVERYTHING_REQUESTED = 'SEND_EVERYTHING_REQUESTED',
+  SEND_EVERYTHING_SUCCEEDED = 'SEND_EVERYTHING_SUCCEEDED',
+  SEND_EVERYTHING_FAILED = 'SEND_EVERYTHING_FAILED',
+  RESET_REQUESTED = 'TRANSACTION_RESET_REQUESTED',
+  RESET_SUCCESSFUL = 'TRANSACTION_RESET_SUCCESSFUL'
+}
+
+export interface SetCurrentValueAction {
+  type: TRANSACTION.CURRENT_VALUE_SET;
+  payload: string;
+}
+
+export interface SetCurrentToAction {
+  type: TRANSACTION.CURRENT_TO_SET;
+  payload: string;
+}
+
+export type CurrentAction = SetCurrentValueAction | SetCurrentToAction;
+
+//#region Send Everything
+export interface SendEverythingRequestedAction {
+  type: TRANSACTION.SEND_EVERYTHING_REQUESTED;
+}
+export interface SendEverythingSucceededAction {
+  type: TRANSACTION.SEND_EVERYTHING_SUCCEEDED;
+}
+export interface SendEverythingFailedAction {
+  type: TRANSACTION.SEND_EVERYTHING_FAILED;
+}
+
+export type SendEverythingAction =
+  | SendEverythingRequestedAction
+  | SendEverythingSucceededAction
+  | SendEverythingFailedAction;
+//#endregion Send Everything
+
+//#region Swap
+export interface SwapTokenToEtherAction {
```

Comment:
Whats the thinking for having these in the same file? lots of duplicate imports?

Reply:
These types have been relegated to their respective subdirectories.